# This file defines joint controller parameter values used for simulating a

chasi:
  # Publish the joint states to joint_states.
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 30

  #left_steering_ctrlr:
  #  joint: left_steering_joint
  #  type: effort_controllers/JointPositionController
  #  pid: {p: 4.0, i: 0.0, d: 1.0}
  #right_steering_ctrlr:
  #  joint: right_steering_joint
  #  type: effort_controllers/JointPositionController
  #  pid: {p: 4.0, i: 0.0, d: 1.0}

  #left_front_axle_ctrlr:
  front_left_wheel_joint_controller:
    joint: front_left_wheel_joint
    type: effort_controllers/JointVelocityController
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  #right_front_axle_ctrlr:
  front_right_wheel_joint_controller:
    joint: front_right_wheel_joint
    type: effort_controllers/JointVelocityController
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  #left_rear_axle_ctrlr:
  rear_left_wheel_joint_controller:
    joint: rear_left_wheel_joint
    type: effort_controllers/JointVelocityController
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}
  #right_rear_axle_ctrlr:
  rear_right_wheel_joint_controller:
    joint: rear_right_wheel_joint
    type: effort_controllers/JointVelocityController
    pid: {p: 1.5, i: 1.0, d: 0.0, i_clamp: 10.0}

  front_left_hinge_joint_controller:
    joint: front_left_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.0, d: 1.0}
    #pid: {p: 4.0, i: 1.0, d: 0.0,  i_clamp: 1}
  front_right_hinge_joint_controller:
    joint: front_right_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.0, d: 1.0}
    #pid: {p: 4.0, i: 1.0, d: 0.0,  i_clamp: 1}
  rear_left_hinge_joint_controller:
    joint: rear_left_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.0, d: 1.0}
    #pid: {p: 4.0, i: 1.0, d: 0.0, i_clamp: 1}
  rear_right_hinge_joint_controller:
    joint: rear_right_hinge_joint
    type: effort_controllers/JointPositionController
    pid: {p: 100.0, i: 0.0, d: 1.0}
    #pid: {p: 4.0, i: 1.0, d: 0.0,  i_clamp: 1}

